streams/src/test/java/org/apache/kafka/streams/integration/EOSUncleanShutdownIntegrationTest.java

```diff
@@ -140,9 +140,6 @@ public void shouldWorkWithUncleanShutdownWipeOutStateStore() throws InterruptedE
             IntegrationTestUtils.produceSynchronously(producerConfig, false, input, Optional.empty(),
                 singletonList(new KeyValueTimestamp<>("k1", "v1", 0L)));
 
-            TestUtils.waitForCondition(stateDir::exists,
-                "Failed awaiting CreateTopics first request failure");
```

Comment:
Can you explain why we need to remove this? It seems like the application must have created the state directory by this point, right?

Reply:
- Previous: The test asserts that the (empty) StateStore directory is not deleted.
- Now: The empty StateStore directory is deleted in the cleanup process, so this assertion is no longer valid. (wait, would it much better to negate the logical condition instead of removing it?)

Reply:
Hmm, I see. Maybe I misread it. I thought we were asserting that Streams actually creates the directory when it starts processing, so that the later assertion (L159) that it cleans up the directory is actually valid.

Then again, L159 seems to assert the opposite thing that its own comment states, so maybe this whole test is just wacky.

The only clue about what we're trying to do here is the name, which almost doesn't make sense at all: `shouldWorkWithUncleanShutdownWipeOutStateStore`. I guess it's saying that we should delete the state store on unclean shutdown? I think that's not what we do anyway. I think we just guarantee that we write an empty checkpoint file. So maybe we should instead change the assertion to:
* the directory is absent
* OR the directory is present, but the checkpoint file is missing
* OR the directory and checkpoint file are present, but the checkpoint file is empty

Any of those should indicate a successful unclean shutdown.